File: src/PythonUnitTests/IteratorTests.py
import unittest
import numpy as np
from nptest import nptest


class IteratorTests(unittest.TestCase):
    def test_nditer_1(self):
      
        a = np.arange(0, 6).reshape(2,3)
        b = np.array([7,8,9])

        for aa in np.nditer(a):
            print(aa)


        for aa in np.nditer((a,b)):
            print(aa)

            
        for aa in np.nditer((a,b,a,b)):
            print(aa)

        #for bb in np.nditer((a,b), flags=['multi_index', 'refs_ok', 'zerosize_ok']).itviews[0]:
        #    print(bb)

        #for cc in np.nditer((a,b), flags=['multi_index', 'refs_ok', 'zerosize_ok']).itviews[1]:
        #    print(cc)


    def test_ndindex_1(self):

        for aa in np.ndindex((2,3)):
            print(aa)

    def test_ndenumerate_1(self):

        a = np.arange(0, 6).reshape(2,3)

        for aa in np.ndenumerate(a):
            print(aa)


if __name__ == '__main__':
    unittest.main()

File: src/NumpyDotNet/UnitTests/NumpyDotNetTests/IteratorTests.cs
﻿using System;
using System.Numerics;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using NumpyDotNet;
using System.Collections.Generic;
using System.Text;
using System.Linq;
using NumpyLib;
#if NPY_INTP_64
using npy_intp = System.Int64;
#else
using npy_intp = System.Int32;
#endif

namespace NumpyDotNetTests
{
    [TestClass]
    public class IteratorTests : TestBaseClass
    {
        [TestMethod]
        public void test_nditer_1()
        {
            var a = np.arange(0, 6).reshape((2, 3));
            var b = np.array(new int[] { 7, 8, 9 });

            foreach (var aa in new nditer(a))
            {
                print(aa);
            }

            foreach (var aa in new nditer((a, b)))
            {
                print(aa);
            }

            foreach (var aa in new nditer((a, b, a, b)))
            {
                print(aa);
            }

        }
 
        [TestMethod]
        public void test_ndindex_1()
        {
            var a = np.arange(0, 6).reshape((2, 3));  // force numpy to be initialized

            foreach (var aa in new ndindex((2, 3)))
            {
                print(aa);
            }


            foreach (var aa in new ndindex((2, 3, 2)))
            {
                print(aa);
            }

            foreach (var aa in new ndindex((3)))
            {
                print(aa);
            }

        }

        [TestMethod]
        public void test_ndenumerate_1()
        {
            var a = np.arange(0, 6).reshape((2, 3));

            foreach (ValueTuple<npy_intp[], object> aa in new ndenumerate(a))
            {
                print(aa.Item1);
                print(aa.Item2);
            }
        }

    }
}
